test/helpers/integration.rb

```diff
@@ -460,7 +460,7 @@ def restart_does_not_drop_connections(
             else
               mutex.synchronize { replies[:unexpected_response] += 1 }
             end
-          rescue Errno::ECONNRESET, Errno::EBADF, Errno::ENOTCONN
+          rescue Errno::ECONNRESET, Errno::EBADF, Errno::ENOTCONN, Errno::ENOTSOCK
```

Comment:
TIL `ENOTSOCK` "Socket operation on non-socket"

Reply:
A CI run of master-yjit raised it.  https://github.com/puma/puma/actions/runs/16813154633/job/47623296437?pr=3687#step:10:877